### PR TITLE
Fix Feature#1609- Lock page navigation when entering data.

### PR DESCRIPTION
**Pull request by tapanhalani**
Signed-off-by: Tapan Halani tapanhalani231@gmail.com

### CODE REVIEW

**Comment by D-storm** [comment]
[exec] There was 1 failure:
     [exec] 
     [exec] 1) PMA_DbSearch_Test::testGetSelectionForm
     [exec] Failed asserting that two strings are equal.
     [exec] --- Expected
     [exec] +++ Actual
     [exec] @@ @@
     [exec] -'<a id="db_search"></a><form id="db_search_form" class="ajax" method="post" action="db_search.php" name="db_search"><input type="hidden" name="db" value="pma" /><input type="hidden" name="lang" value="en" /><input type="hidden" name="collation_connection" value="utf-8" /><input type="hidden" name="token" value="token" /><fieldset><legend>Search in database</legend><table class="formlayout"><tr><td>Words or values to search for (wildcard: "%"):</td><td><input type="text" name="criteriaSearchString" size="60" value="" /></td></tr><tr><td class="right vtop">Find:</td><td><input type="radio" name="criteriaSearchType" id="criteriaSearchType_1" value="1" checked="checked" />
     [exec] +'<a id="db_search"></a><form id="db_search_form" class="ajax lock-page" method="post" action="db_search.php" name="db_search"><input type="hidden" name="db" value="pma" /><input type="hidden" name="lang" value="en" /><input type="hidden" name="collation_connection" value="utf-8" /><input type="hidden" name="token" value="token" /><fieldset><legend>Search in database</legend><table class="formlayout"><tr><td>Words or values to search for (wildcard: "%"):</td><td><input type="text" name="criteriaSearchString" size="60" value="" /></td></tr><tr><td class="right vtop">Find:</td><td><input type="radio" name="criteriaSearchType" id="criteriaSearchType_1" value="1" checked="checked" />
     [exec]  <label for="criteriaSearchType_1">at least one of the words<span class="pma_hint"><span class="hide">Words are separated by a space character (" ").</span></span></label><br />
     [exec]  <input type="radio" name="criteriaSearchType" id="criteriaSearchType_2" value="2" />
     [exec]  <label for="criteriaSearchType_2">all words<span class="pma_hint"><span class="hide">Words are separated by a space character (" ").</span></span></label><br />
     [exec]  <input type="radio" name="criteriaSearchType" id="criteriaSearchType_3" value="3" />
     [exec]  <label for="criteriaSearchType_3">the exact phrase</label><br />
     [exec]  <input type="radio" name="criteriaSearchType" id="criteriaSearchType_4" value="4" />
     [exec]  <label for="criteriaSearchType_4">as regular expression <a href="./url.php?url=http%3A%2F%2Fdev.mysql.com%2Fdoc%2Frefman%2F5.6%2Fen%2Fregexp.html" target="mysql_doc">Documentation</a></label><br />
     [exec]  </td></tr><tr><td class="right vtop">Inside tables:</td><td rowspan="2"><select name="criteriaTables[]" size="6" multiple="multiple"><option value="table1">table1</option><option value="table2">table2</option></select></td></tr><tr><td class="right vbottom"><a href="#" onclick="setSelectOptions('db_search', 'criteriaTables[]', true); return false;">Select All</a> &nbsp;/&nbsp;<a href="#" onclick="setSelectOptions('db_search', 'criteriaTables[]', false); return false;">Unselect All</a></td></tr><tr><td class="right">Inside column:</td><td><input type="text" name="criteriaColumnName" size="60"value="" /></td></tr></table></fieldset><fieldset class="tblFooters"><input type="submit" name="submit_search" value="Go" id="buttonGo" /></fieldset></form><div id="togglesearchformdiv"><a id="togglesearchformlink"></a></div>'
     [exec] 
     [exec] /home/travis/build/phpmyadmin/phpmyadmin/test/classes/PMA_DbSearch_test.php:251

**Comment by tapanhalani** [comment]
Hello D-Storm,

I could not completely understand what is going wrong with DB_Search page. I cannot understand the above failure report. Its the first time I have seen one.
Could you guide me through what is going wrong ? I am sure I can fix it .

**Comment by D-storm** [comment]
A test is failing on the following file and line: `test/classes/PMA_DbSearch_test.php:251`. It says that expected output didn't match the actual output of the testcase.

`[exec] --- Expected`, this means that line starting with `-` is the expected output of the testcase.
`[exec] +++ Actual`, this means that line starting with `+` is the actual output of the testcase.

    [exec] -'<a id="db_search"></a><form id="db_search_form" class="ajax" method="post" action="db_search.php" name="db_search"><input type="hidden" name="db" value="pma" /><input type="hidden" name="lang" value="en" /><input type="hidden" name="collation_connection" value="utf-8" /><input type="hidden" name="token" value="token" /><fieldset><legend>Search in database</legend><table class="formlayout"><tr><td>Words or values to search for (wildcard: "%"):</td><td><input type="text" name="criteriaSearchString" size="60" value="" /></td></tr><tr><td class="right vtop">Find:</td><td><input type="radio" name="criteriaSearchType" id="criteriaSearchType_1" value="1" checked="checked" />

    [exec] +'<a id="db_search"></a><form id="db_search_form" class="ajax lock-page" method="post" action="db_search.php" name="db_search"><input type="hidden" name="db" value="pma" /><input type="hidden" name="lang" value="en" /><input type="hidden" name="collation_connection" value="utf-8" /><input type="hidden" name="token" value="token" /><fieldset><legend>Search in database</legend><table class="formlayout"><tr><td>Words or values to search for (wildcard: "%"):</td><td><input type="text" name="criteriaSearchString" size="60" value="" /></td></tr><tr><td class="right vtop">Find:</td><td><input type="radio" name="criteriaSearchType" id="criteriaSearchType_1" value="1" checked="checked" />
    [exec]  <label for="criteriaSearchType_1">at least one of the words<span class="pma_hint"><span class="hide">Words are separated by a space character (" ").</span></span></label><br />
    [exec]  <input type="radio" name="criteriaSearchType" id="criteriaSearchType_2" value="2" />
    [exec]  <label for="criteriaSearchType_2">all words<span class="pma_hint"><span class="hide">Words are separated by a space character (" ").</span></span></label><br />
    [exec]  <input type="radio" name="criteriaSearchType" id="criteriaSearchType_3" value="3" />
    [exec]  <label for="criteriaSearchType_3">the exact phrase</label><br />
    [exec]  <input type="radio" name="criteriaSearchType" id="criteriaSearchType_4" value="4" />
    [exec]  <label for="criteriaSearchType_4">as regular expression <a href="./url.php?url=http%3A%2F%2Fdev.mysql.com%2Fdoc%2Frefman%2F5.6%2Fen%2Fregexp.html" target="mysql_doc">Documentation</a></label><br />
    [exec]  </td></tr><tr><td class="right vtop">Inside tables:</td><td rowspan="2"><select name="criteriaTables[]" size="6" multiple="multiple"><option value="table1">table1</option><option value="table2">table2</option></select></td></tr><tr><td class="right vbottom"><a href="#" onclick="setSelectOptions('db_search', 'criteriaTables[]', true); return false;">Select All</a> &nbsp;/&nbsp;<a href="#" onclick="setSelectOptions('db_search', 'criteriaTables[]', false); return false;">Unselect All</a></td></tr><tr><td class="right">Inside column:</td><td><input type="text" name="criteriaColumnName" size="60"value="" /></td></tr></table></fieldset><fieldset class="tblFooters"><input type="submit" name="submit_search" value="Go" id="buttonGo" /></fieldset></form><div id="togglesearchformdiv"><a id="togglesearchformlink"></a></div>'

phpMyAdmin uses PHPUnit 4.1.x for testing. See PHPUnit 4.1 [documentation].

[documentation]: https://phpunit.de/manual/4.1/en/index.html

**Comment by tapanhalani** [comment]
Thanks for the reply. That check is failing because I have added a new class in the file for Db_search . In order to solve the error, I have to either remove that class lock-page from DB_Search form, which will result in non-locking of the page while editing its elements, or I have to change the test by adding this class to the test-file also , so that it won't give error further. 

Which one should I do? Does everyone want this feature for DB_search page? 
I don't mind removing it , as there is not much data on this page and would be no extra over-head by the user to remove it. I just added it for extra ease and consistency.

**Comment by D-storm** [comment]
You have to fix the tests.

**Comment by tapanhalani** [comment]
I have uploaded the new commit and it seems it has passed all the tests . Please check it here.

**Comment by lem9** [comment]
Thanks. I have merged this, but please submit a patch to fix this:
https://scrutinizer-ci.com/g/phpmyadmin/phpmyadmin/inspections/0c1d385c-263f-4476-a834-c111252422b1/issues/files/js/ajax.js?status=new&orderField=path&order=asc
